crates/aztec_backend/src/acvm_interop/pwg/merkle.rs

```diff
@@ -378,3 +368,111 @@ fn basic_interop_update() {
+
+#[test]
+fn check_membership() {
+    struct Test<'a> {
+        // Index of the leaf in the MerkleTree
+        index: &'a str,
+        // Returns true if the leaf is indeed a part of the MerkleTree at the specified index
+        result: bool,
+        // The message is used to derive the leaf at `index` by using the specified hash
+        message: Vec<u8>,
+        // If this is true, then before checking for membership
+        // we update the tree with the message at that index
+        should_update_tree: bool,
+        error_msg: &'a str,
+    }
+    // Note these test cases are not independent.
+    // i.e. If you update index 0, then this will be saved for the next test
+    let tests = vec![
+        Test {
+            index : "0",
+            result : true,
+            message : vec![0;64],
+            should_update_tree: false,
+            error_msg : "this should always be true, since the tree is initialised with 64 zeroes"
+        },
+        Test {
+            index : "0",
+            result : false,
+            message : vec![10;64],
+            should_update_tree: false,
+            error_msg : "this should be false, since the tree was not updated, however the message which derives the leaf has changed"
+        },
+        Test {
+            index : "0",
+            result : true,
+            message : vec![1;64],
+            should_update_tree: true,
+            error_msg : "this should be true, since we are updating the tree"
+        },
+        Test {
+            index : "0",
+            result : true,
+            message : vec![1;64],
+            should_update_tree: false,
+            error_msg : "this should be true since the index at 4 has not been changed yet, so it would be [0;64]"
+        },
+        Test {
+            index : "4",
+            result : true,
+            message : vec![0;64],
+            should_update_tree: false,
+            error_msg : "this should be true since the index at 4 has not been changed yet, so it would be [0;64]"
+        },
+    ];
+
+    use tempfile::tempdir;
+    let temp_dir = tempdir().unwrap();
+    let mut tree = MerkleTree::new(3, &temp_dir);
+
+    for test_vector in tests {
+        let index = FieldElement::try_from_str(test_vector.index).unwrap();
+        let index_as_usize: usize = test_vector.index.parse().unwrap();
+
+        let leaf = hash(&test_vector.message);
+
+        let mut root = tree.root();
+        if test_vector.should_update_tree {
+            root = tree.update_message(index_as_usize, &test_vector.message);
+        }
+
+        let hash_path = flatten_path(tree.get_hash_path(index_as_usize));
+        let hash_path_ref = hash_path.iter().collect();
+        let result = MerkleTree::check_membership(hash_path_ref, &root, &index, &leaf);
+        let is_leaf_in_true = result == FieldElement::one();
+
+        assert!(is_leaf_in_true == test_vector.result, "{}", test_vector.error_msg);
+    }
+}
+
+#[test]
+fn insert_leaf() {
```

Comment:
Can we remove this test as it doesn't fail or pass any conditions?

Reply:
Done, I was using it to generate my test data, but I have a separate personal example repo for that now